R/allocation_data_cleaning.R

```diff
@@ -0,0 +1,48 @@
+# Author: Paigan Aspinall
+# Date: 20SEP2023
+# R version: 4.2.2
+
+library(dplyr) #1.1.0
+
+#' Apply coded allocations to every row
+#'
+#' This function applies allocations to all rows for a participant
+#'
+#' @param data A REDcap export dataset.
+#' @param allocation_column Column name where allocation data is stored.
+#' @param allocation_event_name Event name where allocations are assigned to participants.
+
+#' @return A dataframe containing allocations.
+#'
+#'
+#' @importFrom dplyr "%>%"
+#' 
+#' @examples:
+#' Example useage:
+#' dataset <- allocation_data_cleaning(data, "pdallocation", "randomisation_arm_1")
+#' 
+#'
+#' @export
+#'
+
+
+allocation_data_cleaning <- function(data, allocation_column, allocation_event_name) {
+  clean_data <- data
```

Comment:
Avoid making a copy of the input data unnecessarily. Creating a copy of the input data and storing it in the variable clean_data won't be efficient if the dataset is large. You can modify the dataset in place (just do all operations on 'data'), which will save memory and improve efficiency.